indexer/src/main.ts

```diff
@@ -82,112 +82,130 @@ export default async function transform({
 
-  const maybeTrieData: Array<TrieData | null> = (events ?? []).map(
-    ({ transaction, receipt, event }) => {
-      console.log(
-        "🔍 Processing transaction with Starknet hash: ",
-        transaction.meta.hash,
-      );
-      // Can be false if the transaction is not related to a specific instance of the Kakarot contract.
-      // This is typically the case if there are multiple Kakarot contracts on the same chain.
-      const isKakarotTx = isKakarotTransaction(transaction);
-      if (!isKakarotTx) {
-        return null;
-      }
-
-      // Skip if the transaction_executed event contains "eth validation failed".
-      if (ethValidationFailed(event)) {
-        return null;
-      }
-
-      const typedEthTx = toTypedEthTx({ transaction });
-      // Can be null if:
-      // 1. The transaction is missing calldata.
-      // 2. The transaction is a multi-call.
-      // 3. The length of the signature array is different from 5.
-      // 4. The chain id is not encoded in the v param of the signature for a
-      //    Legacy transaction.
-      // 5. The deserialization of the transaction fails.
-      if (!typedEthTx) {
-        return null;
-      }
+  const filteredEvents = (events ?? [])
+    // Can be false if the transaction is not related to a specific instance of the Kakarot contract.
+    // This is typically the case if there are multiple Kakarot contracts on the same chain.
+    .filter((event) => isKakarotTransaction(event.transaction))
+    // Skip if the transaction_executed event contains "eth validation failed".
+    .filter((event) => !ethValidationFailed(event.event))
+    .map((event) => ({
+      event: event,
+      typedEthTx: toTypedEthTx({ transaction: event.transaction }),
+    }))
+    // Can be null if:
+    // 1. The transaction is missing calldata.
+    // 2. The transaction is a multi-call.
+    // 3. The length of the signature array is different from 5.
+    // 4. The chain id is not encoded in the v param of the signature for a
+    //    Legacy transaction.
+    // 5. The deserialization of the transaction fails.
+    .filter((eventExtended) => eventExtended.typedEthTx !== null)
+    .map((eventExtended) => {
       const ethTx = typedTransactionToEthTx({
-        typedTransaction: typedEthTx,
-        receipt,
+        typedTransaction: eventExtended.typedEthTx!,
+        receipt: eventExtended.event.receipt,
         blockNumber,
         blockHash,
         isPendingBlock,
       });
-      // Can be null if:
-      // 1. The typed transaction if missing a signature param (v, r, s).
-      if (!ethTx) {
-        return null;
-      }
 
-      // Can be null if:
-      // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
-      // 2. The event has an invalid number of keys.
-      const ethLogs = receipt.events
-        .map((e) => {
-          return toEthLog({
-            transaction: ethTx,
+      return { ...eventExtended, ethTx };
+    })
```

Comment:
Do you think it might be possible to break the code using helper functions to make it more readable?